packages/fast-components-styles-msft/src/utilities/color/recipe.ts

```diff
@@ -0,0 +1,113 @@
+import {
+    checkDesignSystemResolver,
+    DesignSystem,
+    DesignSystemResolver,
+} from "../../design-system";
+import { backgroundColor } from "../design-system";
+import { Swatch, SwatchFamily } from "./common";
+import {
+    findClosestSwatchIndex,
+    findSwatchIndex,
+    getSwatch,
+    isDarkMode,
+    Palette,
+    swatchByContrast,
+} from "./palette";
+
+/**
+ * Resolves the index that the contrast search algorithm should start at
+ */
+function referenceColorInitialIndexResolver(
+    referenceColor: string,
+    sourcePalette: Palette,
+    designSystem: DesignSystem
+): number {
+    return findClosestSwatchIndex(sourcePalette, referenceColor)(designSystem);
+}
+
+function contrastTargetFactory(
+    targetContrast: number
+): (instanceContrast: number) => boolean {
+    return (instanceContrast: number): boolean => instanceContrast >= targetContrast;
+}
+
+function indexToSwatchFamily(
+    accessibleIndex: number,
+    palette: Palette,
+    direction: number,
+    restDelta: number,
+    hoverDelta: number,
+    activeDelta: number
+): SwatchFamily {
+    // One of the indexes will be rest, the other will be hover. Depends on the offsets and the direction.
+    const accessibleIndex2: number =
+        accessibleIndex + direction * Math.abs(restDelta - hoverDelta);
+
+    const indexOneIsRestState: boolean =
+        direction === 1
+            ? restDelta < hoverDelta
+            : direction * restDelta > direction * hoverDelta;
+
+    const restIndex: number = indexOneIsRestState ? accessibleIndex : accessibleIndex2;
+    const hoverIndex: number = indexOneIsRestState ? accessibleIndex2 : accessibleIndex;
+
+    const activeIndex: number = restIndex + direction * activeDelta;
+
+    return {
+        rest: getSwatch(restIndex, palette),
+        hover: getSwatch(hoverIndex, palette),
+        active: getSwatch(activeIndex, palette),
+    };
+}
+
+/**
+ * Function to derive accessible colors from contrast and delta configuration.
+ * Performs a simple contrast check against the colors and returns
+ * the color that has the most contrast against the background. If contrast
+ * cannot be retrieved correctly, function returns black.
+ */
+export function accessibleAlgorithm(
+    palette: Palette | DesignSystemResolver<Palette>,
+    minContrast: number | DesignSystemResolver<number>,
+    restDelta: number | DesignSystemResolver<number>,
+    hoverDelta: number | DesignSystemResolver<number>,
+    activeDelta: number | DesignSystemResolver<number>
```

Comment:
Do we need to support selected here?

Reply:
I'm actually trying to get rid of selected, so I didn't add it to the new recipe. We currently only use the base neutral fill for selected, and it's complicating a lot of the rules and usage patterns. One of the things removing it would address is the ability to then have hover and active state for a selected element.